File: Solution-Go/Combination_40.go
func combinationSum2(candidates []int, target int) [][]int {
    res := [][]int{}
    sort.Ints(candidates)
    helper(candidates, &res, []int{}, 0, target)
    return res
}

func helper(candidates []int, res *[][]int, tmp []int, start int, target int) {
    if target == 0 {
        b := make([]int, len(tmp))
        copy(b, tmp)
        *res = append(*res, b)
    } else {
        for i := start; i < len(candidates); i++ {
            if candidates[i] > target {
                break
            }
            if i > start && candidates[i] == candidates[i - 1] {
                continue
            }
            tmp = append(tmp, candidates[i])
            helper(candidates, res, tmp, i + 1, target - candidates[i])
            tmp = tmp[:len(tmp) - 1]
        }
    }
}

File: Solution-Python/Combination_40.py
class Combination_40:
    def combinationSum2(self, candidates: List[int], target: int) -> List[List[int]]:
        res = []
        self.helper(sorted(candidates), res, [], 0, target)
        return res
    
    def helper(self, candidates: List[int], res: List[List[int]], tmp: List[int], start: int, target: int):
        if target == 0:
            res.append(copy.deepcopy(tmp))
        else:
            for i in range(start, len(candidates)):
                if candidates[i] > target:
                    break
                if i > start and candidates[i] == candidates[i - 1]:
                    continue
                tmp.append(candidates[i])
                self.helper(candidates, res, tmp, i + 1, target - candidates[i])
                tmp.pop()